### PR TITLE
BMFont working on non-monospace fonts.

### PR DESCRIPTION
**Pull request by MichaeliusDenModige**
Only briefly tested on EB Garamond from http://www.georgduffner.at/ebgaramond/

I have only changed parts where I spotted code differences, so mistakes may occur. I have a full file available in case these changes do not work.

### CODE REVIEW

**Comment by giawa** [comment]
This is awesome!  Thanks so much for putting together this pull request.  My only issue is that the fonts now seem to be vertically displaced by the font height when using the new ch.Bearing value for generating the y value of the vertices.  I think the 'Height' should be added to each y value as the vertices are constructed, which will put the new code in closer alignment with the old code.  What do you think?  This would cause the y-axis to be more in line with where the bottom of a typical character would be (or more in line with where a line would be on a piece of paper), instead of the y-axis being at the top of the font.

Here's a comparison of before and after:  https://imgur.com/a/ggTCnO9

File: OpenGLTutorial10/OpenGLTutorial10/Font.cs [hunk 1]
```diff
@@ -195,16 +190,20 @@ public FontVAO CreateString(ShaderProgram program, string text, Justification ju
                 // grab the character, replacing with ' ' if the character isn't loaded
                 Character ch = characters[characters.ContainsKey(text[(int)i]) ? text[(int)i] : ' '];
 
-                vertices[i * 4 + 0] = new Vector3(xpos, ch.height, 0);
-                vertices[i * 4 + 1] = new Vector3(xpos, 0, 0);
-                vertices[i * 4 + 2] = new Vector3(xpos + ch.width, ch.height, 0);
-                vertices[i * 4 + 3] = new Vector3(xpos + ch.width, 0, 0);
-                xpos += (int)ch.width;
+                vertices[i * 4 + 0] = new Vector3(xpos + ch.bearing.X,- ch.bearing.Y, 0);
```
Comment (giawa) [inline OpenGLTutorial10/OpenGLTutorial10/Font.cs]: Changing this line (and the following lines) to add the Height to the y position will better align the resulting vertices with the previous code.

Comment (MichaeliusDenModige) [reply OpenGLTutorial10/OpenGLTutorial10/Font.cs]: Yes, there seems to be some vertical displacement regardless of font, however I do not know which 'Height' you refer to.
If you meant _ch.size.Y_, then this breaks the vertical alignment of the non-monospace font, as characters can have differing vertical sizes and bearings.
_FontTexture.Size.Height_ adjusts the position by too much in my case (about 500 pixels), as I use a large texture.

Comment (giawa) [reply OpenGLTutorial10/OpenGLTutorial10/Font.cs]: I mean the actual Height property of the BMFont.

`                vertices[i * 4 + 0] = new Vector3(xpos + ch.bearing.X, Height - ch.bearing.Y, 0);
                vertices[i * 4 + 1] = new Vector3(xpos + ch.bearing.X, Height - (ch.bearing.Y + ch.size.Y), 0);
                vertices[i * 4 + 2] = new Vector3(xpos + ch.bearing.X + ch.size.X, Height - ch.bearing.Y, 0);
                vertices[i * 4 + 3] = new Vector3(xpos + ch.bearing.X + ch.size.X, Height - (ch.bearing.Y + ch.size.Y), 0);`

Comment (MichaeliusDenModige) [reply OpenGLTutorial10/OpenGLTutorial10/Font.cs]: My apologies, I did indeed miss it when reviewing your suggestion.
As you asked, I have added the Height to the vertices, and it does align better while still working with non-monospace fonts.

**Comment by giawa** [comment]
Awesome!  Looks good to me.  Thanks for submitting this pull request.

**Pull request merged**